domain/application/state/storage.go

```diff
@@ -118,6 +135,226 @@ VALUES ($storageToAdd.*)`, storageToAdd{})
 	return nil
 }
 
+type storageTemplate struct {
+	meta   charmStorage
+	params application.ApplicationStorageArg
+}
+
+// insertUnitStorage inserts the storage records need to record the intent
+// for the specified new unit and storage args. Records include:
+// - storage instance
+// - filesystem
+// - volume
+// - related attachment records
+// TODO(storage) - support attaching existing storage when adding a unit
+func (st *State) insertUnitStorage(
+	ctx context.Context, tx *sqlair.TX, modelType model.ModelType, storageParentDir string,
```

Comment:
model type should be removed I think. Would you consider breaking all of these on to seperate lines so they're easier to reason about. My eyes go cross eyed reading this :)

Reply:
This is called in a loop. It all needs refactoring. As a compromise till that happens, the model type is looked up outside the loop.